ingest/ledger_transaction.go

```diff
@@ -181,3 +188,405 @@ func (t *LedgerTransaction) operationChanges(ops []xdr.OperationMeta, index uint
+	if minSequenceAge == nil {
+		return 0, false
+	}
+
+	return int64(*minSequenceAge), true
+}
+
+func (t *LedgerTransaction) MinSequenceLedgerGap() (int64, bool) {
+	minSequenceLedgerGap := t.Envelope.MinSeqLedgerGap()
+	if minSequenceLedgerGap == nil {
+		return 0, false
+	}
+
+	return int64(*minSequenceLedgerGap), true
+}
+
+func (t *LedgerTransaction) GetSorobanData() (xdr.SorobanTransactionData, bool) {
+	switch t.Envelope.Type {
+	case xdr.EnvelopeTypeEnvelopeTypeTx:
+		return t.Envelope.V1.Tx.Ext.GetSorobanData()
+	case xdr.EnvelopeTypeEnvelopeTypeTxFeeBump:
+		return t.Envelope.FeeBump.Tx.InnerTx.V1.Tx.Ext.GetSorobanData()
+	default:
+		return xdr.SorobanTransactionData{}, false
+	}
+}
+
+func (t *LedgerTransaction) SorobanResourceFee() (int64, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return int64(sorobanData.ResourceFee), true
+}
+
+func (t *LedgerTransaction) SorobanResourcesInstructions() (uint32, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return uint32(sorobanData.Resources.Instructions), true
+}
+
+func (t *LedgerTransaction) SorobanResourcesReadBytes() (uint32, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return uint32(sorobanData.Resources.ReadBytes), true
+}
+
+func (t *LedgerTransaction) SorobanResourcesWriteBytes() (uint32, bool) {
+	sorobanData, ok := t.GetSorobanData()
+	if !ok {
+		return 0, false
+	}
+
+	return uint32(sorobanData.Resources.WriteBytes), true
+}
+
+func (t *LedgerTransaction) InclusionFeeBid() (int64, bool) {
+	resourceFee, ok := t.SorobanResourceFee()
+	if !ok {
+		return 0, false
+	}
+
+	return int64(t.Envelope.Fee()) - resourceFee, true
+}
+
+func (t *LedgerTransaction) FeeAccountAddress() (string, bool) {
+	switch t.Envelope.Type {
+	case xdr.EnvelopeTypeEnvelopeTypeTx:
+		sourceAccount := t.Envelope.SourceAccount()
+		feeAccountAddress := sourceAccount.Address()
+		return feeAccountAddress, true
+	case xdr.EnvelopeTypeEnvelopeTypeTxFeeBump:
+		feeBumpAccount := t.Envelope.FeeBumpAccount()
+		feeAccountAddress := feeBumpAccount.Address()
+		return feeAccountAddress, true
+	default:
+		return "", false
+	}
+}
+
+func (t *LedgerTransaction) SorobanInclusionFeeCharged() (int64, bool) {
+	resourceFee, ok := t.SorobanResourceFee()
+	if !ok {
+		return 0, false
+	}
+
+	feeAccountAddress, ok := t.FeeAccountAddress()
+	if !ok {
+		return 0, false
+	}
+
+	accountBalanceStart, accountBalanceEnd := getAccountBalanceFromLedgerEntryChanges(t.FeeChanges, feeAccountAddress)
+	initialFeeCharged := accountBalanceStart - accountBalanceEnd
+
+	return initialFeeCharged - resourceFee, true
+}
+
+func (t *LedgerTransaction) InclusionFeeCharged() (int64, bool) {
+	inclusionFee, ok := t.SorobanInclusionFeeCharged()
+	if ok {
+		return inclusionFee, ok
+	}
+
+	// Inclusion fee for classic is just the base 100 stroops + surge pricing if surging
+	inclusionFee, ok = t.FeeCharged()
```

Comment:
I think calculating the inclusion fee is a little more nuanced than this:

```
Fee Charged = inclusion fee * (operation count + if(fee_account is not null, 1, 0)
```

we want to know what that base fee was for the given txn since we can get the Fee charged from `FeeCharged()` function